File: 디자인패턴/java 디자인 패턴/Observer/UnfairDiceGame.java
import java.util.Iterator;
import java.util.Random;

public class UnfairDiceGame extends DiceGame {
    @Override
    public void Play() {
        Random randint = new Random();
        Iterator<Player> iter = players.iterator();
        int dice_Num = randint.nextInt(20)+1;
        if(dice_Num>6) dice_Num = 4;
        System.out.println("Current Dice Num: " + dice_Num);
        while (iter.hasNext()) iter.next().ProcessDiceBetting(dice_Num);
    }
    
}

File: 디자인패턴/Python 디자인 패턴/Observer/UnfairDiceGame.py
from DiceGame import*
import random as rd
class UnfairDiceGame(DiceGame):
    def Play(this):
        diceNum = rd.randrange(20)+1
        if(diceNum > 6): diceNum = 2
        print(f"Current: {diceNum}")
        players = iter(this.Players)
        player = next(players)
        while(player):
            player.Update(diceNum)
            player = next(players, 0)
        pass
    pass